nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -107,6 +112,74 @@ def write_entry(profile, generation, machine_id):
             f.write("machine-id %s\n" % machine_id)
     os.rename(tmp_path, entry_file)
 
+def write_secureboot_entry(profile, generation, machine_id):
+    if profile:
+        entry_file = "@efiSysMountPoint@/loader/entries/nixos-%s-generation-%d.conf" % (profile, generation)
+        efi_file_relative = "EFI/nixos/nixos-%s-generation-%d.efi" % (profile, generation)
+        efi_file = "@efiSysMountPoint@/%s" % (efi_file_relative)
+    else:
+        entry_file = "@efiSysMountPoint@/loader/entries/nixos-generation-%d.conf" % (generation)
+        efi_file_relative = "EFI/nixos/nixos-generation-%d.efi" % (generation)
+        efi_file = "@efiSysMountPoint@/%s" % (efi_file_relative)
+
+    initrd = "%s.initrd.tmp" % (efi_file)
+    shutil.copyfile(
+        profile_path(profile, generation, "initrd"),
+        initrd
+    )
```

Comment:
Is this copy really required when it is copied into `linuxx64.efi.stub`?

Reply:
Ah, ok because of `append-secrets`. Still it would be a good to keep the initrd out of `/boot` to decrease the likely hood of going out of disk-space and because fat32 is not the best filesystem ...

Reply:
with tempfile.TemporaryDirectory() as tmp:
    # here the code goes

Reply:
There could be also a check if `append-secrets` or otherwise we don't need to copy at all.

Reply:
Implemented, check it out!